Finishing up remaining TODOs in panel's demo

b-yogesh opened the pull request:
This PR finishes up the TODOs in both the Hist2D demo panels and Spectrum View.

- **CircularProgress** for Histogram 2D panel: After investigating, it seems infeasible within the current Chartlets architecture. Chartlets executes callbacks sequentially and returns all responses together, preventing timely progress updates. To yield responses as they complete would require changes on the Chartlets.js side. If real-time updates are needed, we may need to rethink the callback execution architecture. As an alternative, skeleton component in Chartlets (currently in development) could be used to indicate progress.

 - **Other TODO**s: All other TODOs for Histogram 2D and Spectrum View have been completed. However, dynamic resizing of the chart when the side panel width changes requires an update to the VegaChart component in Chartlets (https://github.com/bcdev/chartlets/pull/121).
 - **Spectrum View TODO**: https://github.com/xcube-dev/xcube-viewer/pull/512
     Added exploration modes
    - `Update`: Clear the chart but the current selection if any.
    - `Add`: Current spectrum is added and new point selections will be added as new spectra
    - Keeps only last 10 spectra in Add mode(to keep server alive)

Chartlets PR (https://github.com/bcdev/chartlets/pull/122) required to make radio buttons styled for this demo.


 - In addition, this PR also adds the dataset name, time label and place name to the Hist2D plot as required in https://github.com/xcube-dev/xcube-viewer/issues/452 (from https://github.com/xcube-dev/xcube/issues/1134)
Addresses: https://github.com/xcube-dev/xcube/issues/1134

Checklist:

* [ ] ~Add unit tests and/or doctests in docstrings~
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [ ] Test coverage remains or increases (target 100%)

clarasb reviewed (commented):
Review from a user perspective:

1. resizing works great
2. Demopanels: With no available places, the demo panels show the confusing error message, that the panel is disabled. More context might be useful for a user like: `Panel diables due to 0 places available`

![grafik](https://github.com/user-attachments/assets/63cc2533-4059-4477-92f0-18a83c01dda6)

3. Visualisation problem in dark mode:
	- Sidebar: background colour doesn't change, hence the text is not readable
	- some text in the horizontal Controlbar above the map and Colourbar Menu do not change in font colour, therefore the text is not readable either

![grafik](https://github.com/user-attachments/assets/d27b1ed4-ca03-41b3-b619-ad05c95e4311)

4. The seperation in the horizontal menubar looks nice and organizes the menu bar really well, however, the dropdown menus and buttons on the left side are not leveled with the ones on the right side. 

![grafik](https://github.com/user-attachments/assets/6421dff1-9686-44e0-b40a-7f04d222777c)

b-yogesh commented:
Thanks for the user review. Please find my comments below.

> 2\. Demopanels: With no available places, the demo panels show the confusing error message, that the panel is disabled. More context might be useful for a user like: `Panel diables due to 0 places available`

I decided to just convert the print statements to these messages as is. We plan on aligning all these tabs https://github.com/xcube-dev/xcube-viewer/issues/483 so they all have the same UI design which would then also take care of what is missing for each tab to be useful. Regarding the error message, it is of the user's choice, and as mentioned I used the content from the print statements, but I can change it for now.



> Visualisation problem in dark mode:
> 
>     * Sidebar: background colour doesn't adjust to theme, hence the text is not readable
> 
>     * some text in the horizontal Controlbar above the map and Colourbar Menu do not adjust in font colour, therefore the text is not readable either

I am not able to replicate this issue. Please see the screenshot attached.
![Screenshot from 2025-04-07 16-48-09](https://github.com/user-attachments/assets/dbb9fdbf-efbd-473b-bb4c-1228548dc3d3)




> The seperation in the horizontal menubar looks nice and organizes the menu bar really well, however, the dropdown menus and buttons on the left side are not leveled with the ones on the right side.

Unable to replicate this as well. See screenshot above!

b-yogesh commented:
> I am not able to replicate this issue. Please see the screenshot attached.

This issue was being caused by using npm link to use local chartlets build for testing and is not an issue created by this PR. 
PR to fix npm link issue -> https://github.com/xcube-dev/xcube-viewer/pull/516

b-yogesh commented on examples/serve/panels-demo/demo_panels/panel_histo2d.py:
```diff
@@ -27,11 +36,37 @@
+        chart=None,
+        style={
+            "paddingTop": 6,
+            # Since for dynamic resizing we use `container` as width and height for
```
This is a Vega-chart concept which allows the user to set [top-level width or height properties to "container" to indicate that the width or height of the plot should be the same as its surrounding container](https://vega.github.io/vega-lite/docs/size.html#specifying-responsive-width-and-height)

b-yogesh commented:
Thanks for the review Clara.

I have updated the text and fixed the wording as requested.
Regarding other points:


> Plot format:
> 
>     * I prefer the representation of reflectances in point and line format. The bar chart makes it difficult to compare the spectra with each other

This was discussed with Norman to keep the bar plots as line chart does not provide the real information when the distance between the wavelengths is too high.



> * Example: The clearing of the plot when switching from `add`-mode to `update`-mode. This is not what I would expect as a user.

This was done to keep the demo simple.

b-yogesh commented:
> 1. This looks cluttered. Either use a Select widget or have the mode radios in a row. 

Yes, this was already in row mode, but was waiting for the release from chartlets.


> Make **Add** the first mode.

Done


> 2\. Use smaller, secondary typography for the explanation of the panel. Also use markdown widget and don't but the button labels in quotes. Instead use bold letters. Same for the other panels.

Done




> 4. Please move the legend to the bottom so right-side space becomes available:

Done

I assume it was number 3, if you had missed any, let me know. 

Screenshots below showing the panels:
![image](https://github.com/user-attachments/assets/e67c2e55-0c10-4844-8301-45d07a71c8ec)

![image](https://github.com/user-attachments/assets/985e743c-9150-4dda-bc55-0326dd7765f4)

![Screenshot from 2025-06-18 16-13-42](https://github.com/user-attachments/assets/218fe896-d118-4b8c-a5c7-ac84890bde54)

I have also created a PR for xcube-viewer to use the latest Chartlets and pin the versions for `vega` libraries that are defined as peerDeps in chartlets. Pinning is required to avoid dep issue when npm installs the latest version. This might be removed in the future when we dont have dep issues.

Please also review this tiny PR in xcube-viewer: https://github.com/xcube-dev/xcube-viewer/pull/531

forman reviewed (approved):
Thanks @b-yogesh
